internal/bundle/inspect/inspect.go

```diff
@@ -20,39 +20,68 @@ import (
 
 // Info represents information about a bundle.
 type Info struct {
-	Manifest    bundle.Manifest          `json:"manifest,omitempty"`
-	Signatures  bundle.SignaturesConfig  `json:"signatures_config,omitempty"`
-	WasmModules []map[string]interface{} `json:"wasm_modules,omitempty"`
-	Namespaces  map[string][]string      `json:"namespaces,omitempty"`
+	Manifest    bundle.Manifest            `json:"manifest,omitempty"`
+	Signatures  bundle.SignaturesConfig    `json:"signatures_config,omitempty"`
+	WasmModules []map[string]interface{}   `json:"wasm_modules,omitempty"`
+	Namespaces  map[string][]NamespaceInfo `json:"namespaces,omitempty"`
+	Annotations []*ast.AnnotationsRef      `json:"annotations,omitempty"`
 }
 
-func File(path string) (*Info, error) {
-	b, err := loader.NewFileLoader().WithSkipBundleVerification(true).AsBundle(path)
+type NamespaceInfo struct {
+	File          string
+	Title         string
+	Organizations []string
+}
+
+func File(path string, includeAnnotations bool, annotationsFilter []string) (*Info, error) {
+	b, err := loader.NewFileLoader().
+		WithSkipBundleVerification(true).
+		WithProcessAnnotation(true). // Always process annotations, for enriching namespace listing
+		AsBundle(path)
 	if err != nil {
 		return nil, err
 	}
 
 	bi := &Info{Manifest: b.Manifest}
 
-	namespaces := map[string][]string{}
+	namespaces := map[string][]NamespaceInfo{}
+	var modules []*ast.Module
 	for _, m := range b.Modules {
-		namespaces[m.Parsed.Package.Path.String()] = append(namespaces[m.Parsed.Package.Path.String()], filepath.Clean(m.Path))
+		ni := NamespaceInfo{
+			File: filepath.Clean(m.Path),
+		}
+		if pa := ast.FindPackageAnnotations(m.Parsed.Annotations); pa != nil {
+			// Regular annotations overriding not used for namespace info
+			ni.Title = pa.Title
+			ni.Organizations = pa.Organizations
+		}
+		namespaces[m.Parsed.Package.Path.String()] = append(namespaces[m.Parsed.Package.Path.String()], ni)
+		modules = append(modules, m.Parsed)
 	}
 	bi.Namespaces = namespaces
 
+	if includeAnnotations {
+		af, err := toRefs(annotationsFilter)
+		var errs ast.Errors
+		bi.Annotations, errs = ast.GetAnnotations(modules, af...)
+		if len(errs) > 0 {
+			return nil, err
+		}
+	}
+
 	err = bi.getBundleDataWasmAndSignatures(path)
 	if err != nil {
 		return nil, err
 	}
 
-	wasmModules := []map[string]interface{}{}
+	var wasmModules []map[string]interface{}
```

Comment:
for style? fine 🤷 I guess if we touch it, we could also prealloc instead? 🤔 
```suggestion
wasmModules := make([]map[string]interface{}, len(b.WasmModules))
```
(same below for entryPoints)

Reply:
If it's just for style, that's ok, but when might also change https://github.com/open-policy-agent/opa/pull/4388/files#diff-12b370ffa6ce99ee4981245c640f2e5fc69c230ceca5108638397a2b7cc7c875R47

Reply:
This was just to make the linter happy. I'll make it prealloc instead 👍 .